packages/block-editor/src/components/link-control/index.js

```diff
@@ -236,17 +218,18 @@ function LinkControl( {
 						<LinkControlSearchInput
 							value={ inputValue }
 							onChange={ onInputChange }
-							onSelect={ onLinkChange }
+							onSelect={ ( suggestion ) => {
+								setIsEditingLink( false );
+								onChange( { ...value, ...suggestion } );
+							} }
 							renderSuggestions={ renderSearchResults }
 							fetchSuggestions={ getSearchHandler }
 							onReset={ resetInput }
-							onKeyDown={ onKeyDown }
-							onKeyPress={ onKeyPress }
 						/>
 					) }
 
 					{ ! isEditingLink && (
-						<LinkControlSettingsDrawer settings={ currentSettings } onSettingChange={ onSettingsChange } />
+						<LinkControlSettingsDrawer value={ value } settings={ settings } onChange={ onChange } />
```

Comment:
I'm curious as to why we're passing `onChange` into the settings drawer? `onChange` handles the changing of the link whereas `onSettingsChange` was specifically to handle changes to the settings. That said, I'm thinking that in fact both scenarios are infact related to the same entity - the link. This is presumably an effort to normalise things so that there is a single change handler required to manage all changes to the entity of a "link"?

Reply:
exactly, I didn't see any difference between say "title" and "open in a new tab" both are properties of the link. The idea is to only have a single entity, that's the main premise of this refactor.

Reply:
Neat. I must have not been able to see the wood for the trees on this one when I originally worked on it as it seems so obvious now. Thank you